### PR TITLE
Development audio mode

### PR DESCRIPTION
**Pull request by zhemitu-amzn**
### Issue #, if available:
None

### Description of changes:
Disable NoAudio mode, as checked-in presence support will not go into the next GA.

#### Manual test cases (add more as needed):

- [x] Join meeting without CallKit
- [x] Join meeting with CallKit as Incoming
- [x] Join meeting with CallKit as Outgoing
- [x] Leave meeting
- [x] Rejoin meeting
- [x] See metrics
- [x] See attendee presence
- [x] Send audio
- [x] Receive audio
- [x] Mute self
- [x] Unmute self
- [x] See local mute indicator when muted
- [x] See remote mute indicator when other is muted
- [x] Enable and disable local video
- [x] Enable and disable remote video from remote side
- [x] Send local video
- [x] Pause and resume remote video
- [x] Switch local camera
- [x] Receive remote screen share

#### Integration validation (required before release)

- [ ] Unit tests pass
- [ ] Build SDK against simulator with release options
- [ ] Build SDK against device with release options
- [ ] Build SDK against simulator with debug options
- [ ] Build SDK against device with debug options
- [ ] Test Demo against simulator with SDK (debug build) in workspace
- [ ] Test Demo against device with SDK (debug build) in workspace
- [ ] Test DemoObjC against simulator with SDK (debug build) in workspace
- [ ] Test DemoObjC against device with SDK (debug build) in workspace
- [ ] Test Demo against simulator with SDK.framework (release build)
- [ ] Test Demo against device with SDK.framework (release build)
- [ ] Test DemoObjC against simulator with SDK.framework (release build)
- [ ] Test DemoObjC against device with SDK.framework (release build)

### Screenshots, if available:

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: AmazonChimeSDK/AmazonChimeSDK/internal/audio/DefaultAudioClientController.swift [hunk 1]
```diff
@@ -17,7 +17,7 @@ class DefaultAudioClientController: NSObject {
     private let audioClientObserver: AudioClientObserver
     private let audioSession: AudioSession
     private let audioPortOffset = 200
-    private let defaultMicAndSpeaker = false
+    private var muteMicAndSpeaker = false
```
Comment (ziyiz-amzn) [inline AmazonChimeSDK/AmazonChimeSDK/internal/audio/DefaultAudioClientController.swift]: The same comment in Android, use `let` and remove L101?

File: AmazonChimeSDK/AmazonChimeSDKTests/audiovideo/DefaultAudioVideoControllerTests.swift [hunk 2]
```diff
@@ -7,6 +7,7 @@
 //
 
 @testable import AmazonChimeSDK
+@testable import AmazonChimeSDKMedia
```
Comment (ziyiz-amzn) [inline AmazonChimeSDK/AmazonChimeSDKTests/audiovideo/DefaultAudioVideoControllerTests.swift]: What variables/methods are used in this package?

File: AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift [hunk 3]
```diff
@@ -315,6 +315,16 @@ class DefaultAudioClientObserverTests: XCTestCase {
         wait(for: [expect], timeout: defaultTimeout)
     }
 
+    func testAttendeesPresenceChanged_attendeesDidJoinWithoutAudio() {
```
Comment (ziyiz-amzn) [inline AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift]: Should JoinWithoutAudio be included in this PR?

File: AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift [hunk 4]
```diff
@@ -315,6 +315,16 @@ class DefaultAudioClientObserverTests: XCTestCase {
...
+        let attendees = [AttendeeUpdate(profileId: attendeeId, externalUserId: externalUserId, data: 4)]
```
Comment (ziyiz-amzn) [inline AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift]: I have not too much context here, does 4 indicate join with out audio ?

File: AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift [hunk 5]
```diff
@@ -326,6 +336,49 @@ class DefaultAudioClientObserverTests: XCTestCase {
         wait(for: [expect], timeout: defaultTimeout)
     }
 
+    func testAttendeesPresenceChanged_attendeesDidJoinWithoutAudioDuplicate() {
```
Comment (ziyiz-amzn) [inline AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift]: Should this test case be included?

File: AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift [hunk 6]
```diff
@@ -326,6 +336,49 @@ class DefaultAudioClientObserverTests: XCTestCase {
...
+        defaultAudioClientObserver.attendeesPresenceChanged(attendees as [Any])
+        defaultAudioClientObserver.attendeesPresenceChanged(attendees as [Any])
+        let expect = eventually {
+            verify(mockRealTimeObserver.attendeesDidJoinWithoutAudio(attendeeInfo: any())).wasCalled(1)
+        }
+
+        wait(for: [expect], timeout: defaultTimeout)
+    }
+
+    func testAttendeesPresenceChanged_attendeesWithAudioRejoinedWithoutAudio() {
```
Comment (ziyiz-amzn) [inline AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift]: The same as above.

File: AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift [hunk 7]
```diff
@@ -326,6 +336,49 @@ class DefaultAudioClientObserverTests: XCTestCase {
...
+        let expect = eventually {
+            verify(mockRealTimeObserver.attendeesDidJoin(attendeeInfo: any())).wasCalled()
+            verify(mockRealTimeObserver.attendeesDidLeave(attendeeInfo: any())).wasCalled()
+            verify(mockRealTimeObserver.attendeesDidJoinWithoutAudio(attendeeInfo: any())).wasCalled()
+        }
+
+        wait(for: [expect], timeout: defaultTimeout)
+    }
+
+    func testAttendeesPresenceChanged_attendeesWithoutAudioRejoinedWithAudio() {
```
Comment (ziyiz-amzn) [inline AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift]: The same.

File: AmazonChimeSDK/AmazonChimeSDK/internal/audio/DefaultAudioClientController.swift (hunk 1 above)
Comment (zhemitu-amzn) [reply AmazonChimeSDK/AmazonChimeSDK/internal/audio/DefaultAudioClientController.swift]: Done

File: AmazonChimeSDKDemo/AmazonChimeSDKDemo/DeviceSelectionViewController.swift [hunk 8]
```diff
@@ -27,6 +28,10 @@ class DeviceSelectionViewController: UIViewController {
         videoDevicePicker.dataSource = self
         videoFormatPicker.delegate = self
         videoFormatPicker.dataSource = self
+
+        /// Update the visibility of Audio Device selection based on the audio mode
+        updateAudioState(audioEnabled: true)
```
Comment (ziyiz-amzn) [inline AmazonChimeSDKDemo/AmazonChimeSDKDemo/DeviceSelectionViewController.swift]: The same comment in Android, are you supposed to expose audio enable/disable option to builders now?

File: AmazonChimeSDK/AmazonChimeSDKTests/audiovideo/DefaultAudioVideoControllerTests.swift (hunk 2 above)
Comment (zhemitu-amzn) [reply AmazonChimeSDK/AmazonChimeSDKTests/audiovideo/DefaultAudioVideoControllerTests.swift]: Seems none being used. Remove this package.

File: AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift (hunk 3 above)
Comment (zhemitu-amzn) [reply AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift]: It should not. Removed this, as well as some other "join without audio" test cases.

File: AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift (hunk 4 above)
Comment (zhemitu-amzn) [reply AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift]: Yes. I also removed this test case.

File: AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift (hunk 5 above)
Comment (zhemitu-amzn) [reply AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift]: It should not be included. Removed.

File: AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift (hunk 6 above)
Comment (zhemitu-amzn) [reply AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift]: Removed.

File: AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift (hunk 7 above)
Comment (zhemitu-amzn) [reply AmazonChimeSDK/AmazonChimeSDKTests/internal/audio/DefaultAudioClientObserverTests.swift]: Removed.

File: AmazonChimeSDKDemo/AmazonChimeSDKDemo/DeviceSelectionViewController.swift (hunk 8 above)
Comment (zhemitu-amzn) [reply AmazonChimeSDKDemo/AmazonChimeSDKDemo/DeviceSelectionViewController.swift]: Not supposed to. Removed.

File: AmazonChimeSDKDemo/AmazonChimeSDKDemo/data/RosterAttendee.swift [hunk 9]
```diff
@@ -14,11 +14,13 @@ public class RosterAttendee {
     let attendeeName: String?
     var volume: VolumeLevel
     var signal: SignalStrength
+    var attendeeStatus: AttendeeStatus
 
-    init(attendeeId: String, attendeeName: String, volume: VolumeLevel, signal: SignalStrength) {
+    init(attendeeId: String, attendeeName: String, volume: VolumeLevel, signal: SignalStrength, attendeeStatus: AttendeeStatus = AttendeeStatus.joined) {
```
Comment (ziyiz-amzn) [inline AmazonChimeSDKDemo/AmazonChimeSDKDemo/data/RosterAttendee.swift]: For audio mode mono and stereo, do we need to change RosterAttendee constructor here? are we showing audio mode?

File: AmazonChimeSDKDemo/AmazonChimeSDKDemo/RosterTableCell.swift [hunk 10]
```diff
@@ -39,7 +39,11 @@ class RosterTableCell: UITableViewCell {
         indicator.layer.cornerRadius = indicator.frame.size.width / 2.0
         speakLevel.tintColor = .systemGray
 
-        speakLevel.image = getSpeakLevelImage(signal: attendee.signal, volume: attendee.volume)
+        let audioEnabled = attendee.attendeeStatus == .joined
+        speakLevel.isHidden = !audioEnabled
+        if audioEnabled {
+            speakLevel.image = getSpeakLevelImage(signal: attendee.signal, volume: attendee.volume)
+        }
```
Comment (ziyiz-amzn) [inline AmazonChimeSDKDemo/AmazonChimeSDKDemo/RosterTableCell.swift]: Seems `audioEnabled` not required in this PR.

Comment (zhemitu-amzn) [reply AmazonChimeSDKDemo/AmazonChimeSDKDemo/RosterTableCell.swift]: Removed

File: AmazonChimeSDKDemo/AmazonChimeSDKDemo/data/RosterAttendee.swift (hunk 9 above)
Comment (zhemitu-amzn) [reply AmazonChimeSDKDemo/AmazonChimeSDKDemo/data/RosterAttendee.swift]: Reverted the change. The attendee status should always be joined.

File: AmazonChimeSDK/AmazonChimeSDKTests/audiovideo/AudioVideoConfigurationTests.swift [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+        XCTAssertEqual(audioVideoConfigAudioModeMono48K.audioMode, .mono48K)
+        XCTAssertEqual(audioVideoConfigAudioModeMono48K.callKitEnabled, false)
+
+        let audioVideoConfigAudioModeMono16K = AudioVideoConfiguration(audioMode: .mono16K)
+        XCTAssertEqual(audioVideoConfigAudioModeMono16K.audioMode, .mono16K)
+        XCTAssertEqual(audioVideoConfigAudioModeMono16K.callKitEnabled, false)
+
+        let audioVideoConfigCallkit = AudioVideoConfiguration(callKitEnabled: true)
+        XCTAssertEqual(audioVideoConfigCallkit.audioMode, .stereo48K)
+        XCTAssertEqual(audioVideoConfigCallkit.callKitEnabled, true)
```
Comment (achronop) [inline AmazonChimeSDK/AmazonChimeSDKTests/audiovideo/AudioVideoConfigurationTests.swift:32]: You do not test the last ctor that set both params.

**Review by achronop — APPROVED** [review]
Looks good. One small comment that it is tested indirectly later, anyway.

**Review by nggandhi — APPROVED** [review]

File: AmazonChimeSDK/AmazonChimeSDKTests/audiovideo/AudioVideoConfigurationTests.swift (hunk 11 above)
Comment (zhemitu-amzn) [reply AmazonChimeSDK/AmazonChimeSDKTests/audiovideo/AudioVideoConfigurationTests.swift:32]: Will add this later when there is a chance.

File: AmazonChimeSDKDemo/AmazonChimeSDKDemo/RosterTableCell.swift [hunk 12]
```diff
@@ -38,8 +38,6 @@ class RosterTableCell: UITableViewCell {
         indicator.accessibilityIdentifier = isActiveSpeaker ? "\(attendee.attendeeName ?? "") Active" : ""
         indicator.layer.cornerRadius = indicator.frame.size.width / 2.0
         speakLevel.tintColor = .systemGray
-
-        speakLevel.image = getSpeakLevelImage(signal: attendee.signal, volume: attendee.volume)
```
Comment (ziyiz-amzn) [inline AmazonChimeSDKDemo/AmazonChimeSDKDemo/RosterTableCell.swift:42]: Why deleted?

Comment (zhemitu-amzn) [reply AmazonChimeSDKDemo/AmazonChimeSDKDemo/RosterTableCell.swift:42]: That's a mistake. Fixed now.

**Pull request merged**